metadata_finds/metadata_finds.sh

```diff
@@ -0,0 +1,209 @@
+#host="localhost"
+#creating the folder "/epts_metadata_finds"  in home user  directory if  it  does not exist
+echo "Creating the  'epts_metadata_finds' folder  in home user directory if  it  does not exist........"
+if [ ! -d "~/epts_metadata_finds" ];
+	then 
+		mkdir ~/epts_metadata_finds
+fi
+
+#wipe all data in the folder before running
+echo "wiping all data in the folder before running..."
+rm  -rf ~/epts_metadata_finds/*
+
+
+echo "Choosing the running  approach..."
+echo "Do you  want to  perform this  script in all databases? [y/n] (lower case please)"
+read all_db_anserwer
+
+#declaring an empty array to store the databases names 
+declare -a  db_list=()
+
+
+if [ $all_db_anserwer == "y" ]
+	then
+                #write all the out put to a file for show database
+    		mysql -u $mysql_user -p$mysql_passwd -h $host -e "show databases;" >  temp_file.txt
+
+                #readarray db_list   < temp_file.txt
+		
+                #start putting all the data to an array
+                idx=0
+                file="temp_file.txt"
+                while IFS= read -r line;
+                do
+                        echo "Do you  want to  add $line database to the investigation? [y/n] (lower case please)"
+                        read sim </dev/tty
+                        if [[ $sim == "y" ]];
+                                then
+                                        echo "Adding..."
+                                        db_list[idx]=$line;
+                                        idx=$((idx+1)) 
+                        fi
+                done < "$file"
+
+                #unset all default databases from array
+		i_schm="information_schema"
+                p_schm="performance_schema"
+		msql="mysql"
+		daBase="Database"
+                sys="sys"
+                new_schema="new_schema"
+
+		index=0
+		for i in ${db_list[@]}; do
+                        if [ "$i" = "$i_schm" ] || [ "$i" = "$p_schm" ] || [ "$i" = "$msql" ] || [ "$i" = "$daBase" ] || [ "$i" = "$sys" ] || [ "$i" = "$new_schema" ]; then
+                                unset db_list[$index]            
+                        fi 
+                        let index++
+                done
+
+elif [ $all_db_anserwer == "n" ];
+	then
+    		#Loading  the databases names
+   		echo  "How mamy data bases you want to  identify the metadata?"
+    		read db_num
+
+    		index=1
+    
+    		while [ $index -le $db_num ]
+    		do
+   			echo "Type the database name$index: "
+    			read database
+    			db_list[index]=$database;
+    			index=$((index+1)) 
+    		done
+
+else
+    	echo "Please type y or n"
+	exit 1
+fi    
+echo "Here is the list I will be working on ${db_list[@]}"
+echo ""
+echo "Do you which to continue... [y/n] (lower case please)"
+read continuar
+
+if [ $continuar == "n" ]
+        then
+                exit 1
+fi
+declare -A queries
+
+####  the bellow  queries  will used to run in mentioned partners databases  ####
+#--Get List of all programs
```

Comment:
Are we missing concepts here?

Reply:
I was not thinking on getting them with a believe that all partners except ICAP have the same concept dictionary as been proven with latest forms.... You think we need to get them?

Reply:
Sure but shouldnt we still check that this is in fact the case?

Reply:
I've added the concepts to the script, jsut in case we need them :)